### PR TITLE
refactor(ButtonGroup): migrate ButtonGroup to TypeScript

### LINKED ISSUE #922: ButtonGroup Typescript Migration
### TypeScript here we come!
We are migrating to typescript, we are doing so component by component

We would love some help in converting some of our component to typescript - we've created a README
https://github.com/mondaycom/monday-ui-react-core/blob/master/TYPESCRIPT_MIGRATION.md

in this PR we expect you to convert the following files:

- [ ] ButtonGroup.jsx
- [ ] ButtonGroupConstants.js
- [ ] ButtonWrapper.jsx
- [ ] buttonGroup-snapshot-tests.jest.js
- [ ] buttonGroup-tests.jest.js

Good Luck

### PR DESCRIPTION
**Pull request by aayushbisen**
- Migrated ButtonGroup,  ButtonGroupConstants & ButtonWrapper to TypeScript
- Migrated buttonGroup-snapshot-tests and buttonGroup-tests to TypeScript
- Refactor buttonGroup stories as per docs

Closes #922

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [x] PR has description.
- [x] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [x] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [ ] Stories include all flows of using the component.
#### Tests
- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: src/components/ButtonGroup/ButtonGroup.tsx [hunk 1]
```diff
@@ -131,7 +166,7 @@ const ButtonGroup = forwardRef(
 ButtonGroup.sizes = Button.sizes;
 ButtonGroup.kinds = Button.kinds;
 
-ButtonGroup.propTypes = {
+/* ButtonGroup.propTypes = {
```
Comment (SergeyRoyt) [inline src/components/ButtonGroup/ButtonGroup.tsx:169]: Please, remove the commented propTypes

File: src/components/ButtonGroup/ButtonGroup.tsx [hunk 2]
```diff
@@ -1,33 +1,68 @@
...
+      value = "",
+      onSelect = undefined,
+      size = BASE_SIZES.SMALL,
+      kind = ButtonType.SECONDARY,
+      groupAriaLabel = "",
+      tooltipPosition = undefined,
+      tooltipHideDelay = undefined,
+      tooltipShowDelay = undefined,
+      tooltipContainerSelector = undefined,
+      tooltipMoveBy = undefined
```
Comment (SergeyRoyt) [inline src/components/ButtonGroup/ButtonGroup.tsx:65]: No need to specify `undefined` as default value, but not a blocker at all

File: src/components/ButtonGroup/ButtonGroup.tsx [hunk 3]
```diff
@@ -1,33 +1,68 @@
 import React, { useRef, forwardRef, useCallback, useMemo, useEffect, useState } from "react";
 import PropTypes from "prop-types";
```
Comment (SergeyRoyt) [inline src/components/ButtonGroup/ButtonGroup.tsx:2]: This can be removed now

File: src/components/ButtonGroup/ButtonGroup.tsx [hunk 4]
```diff
@@ -1,33 +1,68 @@
...
 import { ButtonWrapper } from "./ButtonWrapper";
 import "./ButtonGroup.scss";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import { BASE_SIZES, SIZES } from "../../constants/sizes";
+import { ButtonType, Size } from "../Button/ButtonConstants";
 
-const ButtonGroup = forwardRef(
+type ButtonGroupOption = {
+  icon?: string | React.FunctionComponent | null;
+  leftIcon?: string | React.FunctionComponent | null;
```
Comment (SergeyRoyt) [inline src/components/ButtonGroup/ButtonGroup.tsx:17]: Please change type of both icons to `IconType`

**Comment by SergeyRoyt** [comment]
Thanks for the great work! Just fix a few notes and we are ready to go :)

**Comment by aayushbisen** [comment]
Thanks for the review @SergeyRoyt 
I have fixed the changes mentioned in notes

File: src/components/ButtonGroup/ButtonGroup.tsx (hunk 4 above)
Comment (aayushbisen) [reply src/components/ButtonGroup/ButtonGroup.tsx]: I totally missed this 😅 
Fixed the type.

File: src/components/ButtonGroup/ButtonWrapper.tsx [hunk 5]
```diff
@@ -1,8 +1,18 @@
...
 import { HIDE_SHOW_EVENTS } from "../Dialog/consts/dialog-show-hide-event";
+import VibeComponentProps from "../../types/VibeComponentProps";
+
+interface ButtonWrapperProps extends VibeComponentProps {
+  tooltipContent: string;
+  tooltipPosition: string;
+  tooltipHideDelay: number;
+  tooltipShowDelay: number;
+  tooltipContainerSelector: string;
+  tooltipMovedBy: { main: number, secondary: number};
```
[the remark this answers is not in the document]
Comment (aayushbisen) [reply src/components/ButtonGroup/ButtonWrapper.tsx]: renamed the prop to tooltipMoveBy.

File: src/components/ButtonGroup/ButtonGroup.tsx (hunk 2 above)
Comment (aayushbisen) [reply src/components/ButtonGroup/ButtonGroup.tsx]: Noted. Removed the default undefined props in the component.

File: src/components/ButtonGroup/ButtonGroup.tsx (hunk 1 above)
Comment (aayushbisen) [reply src/components/ButtonGroup/ButtonGroup.tsx]: Removed the commented propTypes.

**Review by hadasfa — COMMENTED** [review]
Thank you so much for your contribution!

**Pull request merged**